ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/PullBandwidthThrottleIntegrationTest.java

```diff
@@ -0,0 +1,221 @@
+import io.confluent.ksql.serde.FormatFactory;
+import static io.confluent.ksql.test.util.AssertEventually.assertThatEventually;
+import io.confluent.ksql.test.util.EmbeddedSingleNodeKafkaCluster;
+import static io.confluent.ksql.test.util.EmbeddedSingleNodeKafkaCluster.VALID_USER2;
+import io.confluent.ksql.test.util.secure.ClientTrustStore;
+import io.confluent.ksql.test.util.secure.Credentials;
+import io.confluent.ksql.test.util.secure.SecureKafkaHelper;
+import static io.confluent.ksql.util.KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG;
+import static io.confluent.ksql.util.KsqlConfig.KSQL_QUERY_PULL_MAX_HOURLY_BANDWIDTH_CONFIG;
+import static io.confluent.ksql.util.KsqlConfig.KSQL_STREAMS_PREFIX;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PageViewDataProvider;
+import io.confluent.ksql.util.VertxCompletableFuture;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import kafka.zookeeper.ZooKeeperClientException;
+import org.apache.kafka.streams.StreamsConfig;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+import org.junit.After;
+import org.junit.AfterClass;
+import static org.junit.Assert.assertEquals;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+
+@Category({IntegrationTest.class})
+public class PullBandwidthThrottleIntegrationTest {
+
+    private static final PageViewDataProvider TEST_DATA_PROVIDER = new PageViewDataProvider();
+    private static final String TEST_TOPIC = TEST_DATA_PROVIDER.topicName();
+    private static final String TEST_STREAM = TEST_DATA_PROVIDER.sourceName();
+
+    private static final String AGG_TABLE = "AGG_TABLE";
+    private static final Credentials NORMAL_USER = VALID_USER2;
+
+    private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.builder()
+            .withKafkaCluster(
+                    EmbeddedSingleNodeKafkaCluster.newBuilder()
+                            .withoutPlainListeners()
+                            .withSaslSslListeners()
+            ).build();
+
+    private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+            .builder(TEST_HARNESS::kafkaBootstrapServers)
+            .withProperty("security.protocol", "SASL_SSL")
+            .withProperty("sasl.mechanism", "PLAIN")
+            .withProperty("sasl.jaas.config", SecureKafkaHelper.buildJaasConfig(NORMAL_USER))
```

Comment:
Just FYI, some of these jaas configs aren't strictly necessary for this test, though it doesn't hurt.